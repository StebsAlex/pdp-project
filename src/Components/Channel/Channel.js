import React from "react";

const Channel = ({ channel, addMe }) => {
  const handleClick = () => addMe(channel);
  return (
    <li data-test="component-news">
      <p> {channel.name}</p>
      <button className="btn btn-small" onClick={handleClick}>
        {" "}
        Add me!{" "}
      </button>
    </li>
  );
};

export default Channel;
