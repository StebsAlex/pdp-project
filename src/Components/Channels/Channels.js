import React from "react";
import Channel from "../Channel/Channel";

const Channels = ({ channels, addMe, title }) => (
  <div className="row">
    <h5>{title} Channel</h5>
    <ul>
      {channels.map(channel => (
        <Channel key={channel.id} channel={channel} addMe={addMe} />
      ))}
    </ul>
  </div>
);

export default Channels;
