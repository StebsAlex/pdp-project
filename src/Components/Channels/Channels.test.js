import React from "react";
import { shallow } from "enzyme";
import Channels from "./Channels";
import Channel from "../Channel/Channel";

const defaultProps = {
  channels: [{ id: 1, name: "blah" }],
  title: "News"
};

describe("Channels component", () => {
  const render = props => shallow(<Channels {...defaultProps} {...props} />);

  test("it renders title and Channel components", () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });

  test("it sets the addMe prop on the Channel", () => { // something snapshot won't do!
    const addMe = jest.fn()
    const wrapper = render({ addMe });
    const channel = wrapper.find(Channel).first();

    expect(channel.prop("addMe")).toEqual(addMe);
  });
});
