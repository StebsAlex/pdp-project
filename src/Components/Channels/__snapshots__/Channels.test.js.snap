// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channels component it renders title and Channel components 1`] = `
<div
  className="row"
>
  <h5>
    News
     Channel
  </h5>
  <ul>
    <Channel
      channel={
        Object {
          "id": 1,
          "name": "blah",
        }
      }
      key="1"
    />
  </ul>
</div>
`;
